armadillo/src/main/java/at/favre/lib/armadillo/SharedPreferenceChangeListenerWrapper.java

```diff
@@ -43,14 +47,14 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
 
         OnSecurePreferenceChangeListener wrapped = wrappedRef.get();
 
-        if (wrapped == null && sharedPreferences != null) {
+        if (wrapped == null) {
             // we unregister ourselves from client preferences
             sharedPreferences.unregisterOnSharedPreferenceChangeListener(this);
             return;
         }
 
-        if (sharedPreferences != null && key != null) {
-            wrapped.onSecurePreferenceChanged(keyComparison, sharedPreferences, key);
+        if (key != null) {
+            wrapped.onSecurePreferenceChanged(securedPrefs, new KeyComparisonImpl(encryptionProtocol, key));
```

Comment:
See comment above. I think it would be better to pass to:

    wrapped.onSecurePreferenceChanged(this, new KeyComparisonImpl(encryptionProtocol, key));

Reply:
I'm not sure if I'm understanding correctly. I'm just passing the `securedPrefs` instance, with is actually initialised with the library `SecureSharedPreferences` instance, so that as you say in your comment the user doesn't have access to the raw encrypted data, but to the wrapper that will allow to get decrypted info.

Reply:
yes, I m not sure it is of much use to get the encrypted data, now would it?

Reply:
Well, but actually the user will be able to access the decrypted data through armadillo's SecureSharedPreferences instance.

What is exactly the change you're proposing here?